sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -207,6 +207,25 @@ public void finish(@Nullable SpanStatus status) {
     this.finishStatus = FinishStatus.finishing(status);
     if (!root.isFinished() && (!waitForChildren || hasAllChildrenFinished())) {
       root.finish(finishStatus.spanStatus);
+
+      // finish unfinished children
+      for (Span child : children) {
+        if (!child.isFinished()) {
+          final Date rootTimestamp = root.getTimestamp();
+          if (rootTimestamp != null) {
+            child.finish(SpanStatus.DEADLINE_EXCEEDED, rootTimestamp);
+          } else {
+            hub.getOptions()
+                .getLogger()
+                .log(
+                    SentryLevel.WARNING,
+                    "Root span - op: %s, description: %s - has no timestamp set, when finishing unfinished spans.",
+                    root.getOperation(),
+                    root.getDescription());
```

Comment:
logging is fine but we still have to `finish` the `child` span with the current time if not `rootTimestamp`, otherwise the whole transaction would be dropped